build-system/tasks/dep-check.js

```diff
@@ -75,7 +75,7 @@ function Rule(config) {
   /** @private @const {!GlobsDef} */
   this.mustNotDependOn_ = toArrayOrDefault(config.mustNotDependOn, []);
 
-  /** @private @const {!GlobsDef} */
+  /** @private @const {!Array<string>} */
```

Comment:
It is generally a good idea. Unfortunately (last I checked) CC does not preserve it into error messages.

As a trivia: You can actually "sub class" string in CC syntax. That gives you a type that can be passed to things that want string, but if you are explicit about it, you cannot pass in strings. The only way to make "instances" is to cast them (which can be controlled with a presubmit).